### PR TITLE
Mr 2 gemtc data

### PR DESCRIPTION
**Pull request by BalancedmonkeY**

### CODE REVIEW

**Review by JanionNevill — CHANGES_REQUESTED** [review]

File: R/DataWrangling.R [hunk 1]
```diff
@@ -39,6 +39,31 @@ CleanData <- function(data) {
...
+      select(starts_with(c("T","N","Mean","SD")))
+  } else {
+    change_cols <- wide_data %>%
+      select(starts_with(c("T","R","N")))
+  }
+  # Transform to long
+  long_data <- wide_data %>%
+                  tidyr::pivot_longer(cols = names(change_cols),
+                                      names_to = c(".value", "arm"),
+                                      names_pattern = "(.*).(.)",
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/DataWrangling.R:60]: Thanks - altered.

File: R/DataWrangling.R [hunk 2]
```diff
@@ -39,6 +39,31 @@ CleanData <- function(data) {
...
+                                      values_drop_na = TRUE
+  )
+  long_data <- long_data %>% dplyr::relocate(FindCovariateNames(long_data), .after = last_col())
+  return(as.data.frame(long_data[,names(long_data)!="arm"]))
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/DataWrangling.R:64]: Thanks - the addition of 'arm' was a workaround, so glad to remove that step

File: R/gemtc_analysis.R [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# User-written functions to wrap up functionality related to running models with {gemtc} #
+
+
+#' Function for formatting standard app user data ready for start of {gemtc} chain of commands
+#' 
+#' @param data Uploaded data from the user (or in-built datasets)
+#' @param outcome_type Indicator of whether data is binary or continuous
+#' @param covariate Chosen covariate name as per uploaded data
+#' @param cov_friendly Friendly version of chosen covariate
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/gemtc_analysis.R:9]: Amended

File: R/gemtc_analysis.R [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+#' @return list containing two dataframes: 1 of core data, another for covariate data
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/gemtc_analysis.R:10]: Added

File: R/gemtc_analysis.R [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+                          responders=long_data$R,
+                          sampleSize=long_data$N)
+  }
+  # specify study level data
+  studyData <- unique(data.frame(study=long_data$Study,
+                                 cov_name = long_data[,covariate]))
+  names(studyData)[2] <- cov_friendly
+  rownames(studyData) <- NULL
+  
+  return(list(armData=armData,studyData=studyData))
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/gemtc_analysis.R:38]: Added spaces throughout all new functions etc.

File: R/DataWrangling.R [hunk 6]
```diff
@@ -39,6 +39,31 @@ CleanData <- function(data) {
...
+#' @param outcome_type Indicator whether outcome is binary or continuous
+#' @return Data frame in long format
+WideToLong <- function(wide_data, outcome_type) {
+  # Specify columns that contain wide data
+  if (outcome_type == "Continuous") {
+    change_cols <- wide_data %>%
+      select(starts_with(c("T","N","Mean","SD")))
+  } else {
+    change_cols <- wide_data %>%
+      select(starts_with(c("T","R","N")))
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/DataWrangling.R:54]: Done

File: R/DataWrangling.R (hunk 2 above)
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply R/DataWrangling.R:64]: Line removed

File: tests/testthat/test-gemtc_analysis.R [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+  expected_data <- list(
+    armData = data.frame(
+      study=c(rep("Constantine",3), rep("Leo",3), rep("Justinian",2)),
+      treatment=c("the Great", "the Younger", "the Dung-named", "the Little", "the Great", "the Butcher", "the Great", "the Slit-nosed"),
+      responders=30:37,
+      sampleSize=100:107),
+    studyData = data.frame(
+      study=c("Constantine", "Leo", "Justinian"),
+      age=rep(99,3))
+  )
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply tests/testthat/test-gemtc_analysis.R:15]: Done

File: tests/testthat/test-gemtc_analysis.R [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+  
+  expect_equal(gemtc_data, expected_data)
```
[the remark this answers is not in the document]
Comment (BalancedmonkeY) [reply tests/testthat/test-gemtc_analysis.R:17]: Good to know - I think I have done this now?

**Review by JanionNevill — APPROVED** [review]

**Pull request merged**